pool/pgpoolstorage/pgpoolstorage.go

```diff
@@ -414,16 +414,16 @@ func (p *PostgresPoolStorage) SetGasPrice(ctx context.Context, gasPrice uint64)
 	return nil
 }
 
-// DeleteGasPricesHistory deletes all gas prices except the last one
-func (p *PostgresPoolStorage) DeleteGasPricesHistory(ctx context.Context) error {
+// DeleteGasPricesHistoryOlderThan deletes all gas prices older than the given date except the last one
+func (p *PostgresPoolStorage) DeleteGasPricesHistoryOlderThan(ctx context.Context, date time.Time) error {
 	sql := `DELETE FROM pool.gas_price
-		WHERE item_id NOT IN (
+		WHERE timestamp < $1 AND item_id NOT IN (
 			SELECT item_id
 			FROM pool.gas_price
 			ORDER BY item_id DESC
 			LIMIT 1
 		)`
```

Comment:
Are you using this query in this way to always leave at least the latest result in the table and not leave it empty, right??

Reply:
Yes